tests/copp/copp_utils.py

```diff
@@ -288,12 +288,16 @@ def configure_always_enabled_for_trap(dut, trap_id, always_enabled):
         always_enabled (str): true or false
     """
     copp_trap_config_json = "/tmp/copp_{}.json".format(trap_id)
+    # Need to remove ip2me trap to test no trap condition to avoid packets being trapped by ip2me after bgp trap removal
```

Comment:
configure_always_enabled_for_trap is been used by many tests. 

How about apply change to `uninstall_trap` only and only do this when user trying to uninstall `bgp`?

Reply:
I have verified that `configure_always_enabled_for_trap` is not being used outside of `copp` tests

So, instead of applying change to `uninstall_trap`, It is better to add a condition `trap_id=="bgp" and always_enabled is false` to `configure_always_enabled_for_trap`?
This will satisfy the check for uninstall bgp and any other trap uninstall scenario

Reply:
We cannot read such implicit message from the function name.
I suggest keep the method as is and do more thing during test preparation:

test trap removal:
```
uninstall_trap(bgp)
uninstall_trap(ip2me) # remove ip2me because bgp traffic can fall back to ip2me trap then interfere following traffic tests
```

test always enable:
```
configure_always_enabled_for_trap(bgp)
uninstall_trap(ip2me) # remove ip2me because bgp traffic can fall back to ip2me trap then interfere following traffic tests
```